lib/cppcheck.cpp

```diff
@@ -391,11 +391,11 @@ static void createDumpFile(const Settings& settings,
     fdump << "<dumps" << language << ">\n";
     fdump << "  <platform"
           << " name=\"" << settings.platform.toString() << '\"'
-          << " char_bit=\"" << settings.platform.char_bit << '\"'
-          << " short_bit=\"" << settings.platform.short_bit << '\"'
-          << " int_bit=\"" << settings.platform.int_bit << '\"'
-          << " long_bit=\"" << settings.platform.long_bit << '\"'
-          << " long_long_bit=\"" << settings.platform.long_long_bit << '\"'
+          << " char_bit=\"" << static_cast<unsigned>(settings.platform.char_bit) << '\"'
```

Comment:
`uint8_t` is treated as a character with stream insertion operators. This caused invalid dump files to be written.

First, we should not write these files raw but using TinyXML2.

Second, I wonder if it would make sense to have a check which detects the usage of `uint8_t` in stream insertion operators. If you want it to a be an actual character you should probably use `{unsigned|signed} char` or `char8_t`.

Reply:
> Second, I wonder if it would make sense to have a check which detects the usage of uint8_t in stream insertion operators. If you want it to a be an actual character you should probably use {unsigned|signed} char or char8_t.

Spontanously that checker sounds reasonable to me.